File: backend/src/main/java/com/g4ts/agendaapp/model/Post.java
package com.g4ts.agendaapp.model;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "Post")
@Getter @Setter @ToString @Builder
@AllArgsConstructor
@NoArgsConstructor
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String titulo;
    private String contenido;
    private LocalDate fecha;

    @OneToOne
    @JoinColumn(name = "idPlugin")
    private Plugin plugin;

    @OneToOne
    @JoinColumn(name = "idEditor")
    private Usuario usuario;
}

File: frontend/src/app/model/post.ts
import { Plugin } from "./plugin";
import { Usuario } from "./usuario";

export class Post {
    id: string;
    plugin: Plugin;
    titulo: string;
    contenido: string;
    fecha: string;
    usuario: Usuario;

    constructor() {
        this.plugin = new Plugin();
        this.usuario = new Usuario();
    }
}
